code/modules/surgery/bodyparts/_bodyparts.dm

```diff
@@ -707,124 +779,86 @@
+//set is_creating to true if you want to change the appearance of the limb outside of mutation changes or forced changes.
+/obj/item/bodypart/proc/update_limb(dropping_limb = FALSE, is_creating = FALSE)
+	SHOULD_CALL_PARENT(TRUE)
 
-	if(owner)
-		owner.updatehealth()
-		owner.update_body() //if our head becomes robotic, we remove the lizard horns and human hair.
-		owner.update_hair()
-		owner.update_damage_overlays()
+	if(HAS_TRAIT(owner, TRAIT_HUSK) && IS_ORGANIC_LIMB(src))
+		dmg_overlay_type = "" //no damage overlay shown when husked
+		is_husked = TRUE
+	else
+		dmg_overlay_type = initial(dmg_overlay_type)
+		is_husked = FALSE
 
-/obj/item/bodypart/proc/is_organic_limb()
-	return (status == BODYPART_ORGANIC)
+	if(!dropping_limb && owner.dna?.check_mutation(/datum/mutation/human/hulk)) //Please remove hulk from the game. I beg you.
+		mutation_color = "#00aa00"
+	else
+		mutation_color = null
 
-//we inform the bodypart of the changes that happened to the owner, or give it the informations from a source mob.
-/obj/item/bodypart/proc/update_limb(dropping_limb, mob/living/carbon/source)
-	var/mob/living/carbon/limb_owner
-	if(source)
-		limb_owner = source
-		if(!original_owner)
-			original_owner = WEAKREF(source)
+	if(mutation_color) //I hate mutations
+		draw_color = mutation_color
+	else if(should_draw_greyscale)
+		draw_color = (species_color) || (skin_tone && skintone2hex(skin_tone))
 	else
-		limb_owner = owner
-		if(original_owner && !IS_WEAKREF_OF(owner, original_owner)) //Foreign limb
-			no_update = TRUE
-		else
-			no_update = FALSE
+		draw_color = null
 
-	if(HAS_TRAIT(limb_owner, TRAIT_HUSK) && is_organic_limb())
-		species_id = "husk" //overrides species_id
-		dmg_overlay_type = "" //no damage overlay shown when husked
-		should_draw_gender = FALSE
-		should_draw_greyscale = FALSE
-		no_update = TRUE
-
-	if(HAS_TRAIT(src, TRAIT_PLASMABURNT) && is_organic_limb())
-		species_id = SPECIES_PLASMAMAN
-		dmg_overlay_type = ""
-		should_draw_gender = FALSE
-		should_draw_greyscale = FALSE
-		no_update = TRUE
-
-	if(HAS_TRAIT(limb_owner, TRAIT_INVISIBLE_MAN) && is_organic_limb())
-		species_id = "invisible" //overrides species_id
-		dmg_overlay_type = "" //no damage overlay shown when invisible since the wounds themselves are invisible.
-		should_draw_gender = FALSE
-		should_draw_greyscale = FALSE
-		no_update = TRUE
-
-	if(no_update)
+	if(!is_creating)
 		return
 
-	if(!animal_origin)
-		var/mob/living/carbon/human/human_owner = limb_owner
-		should_draw_greyscale = FALSE
+	if(!animal_origin && ishuman(owner))
+		var/mob/living/carbon/human/human_owner = owner
 
 		var/datum/species/owner_species = human_owner.dna.species
-		species_id = owner_species.limbs_id
-		species_flags_list = human_owner.dna.species.species_traits
+		species_flags_list = human_owner.dna.species.species_traits //Literally only exists for a single use of NOBLOOD, but, no reason to remove it i guess...?
+		limb_gender = (human_owner.gender == MALE) ? "m" : "f"
```

Comment:
tg code uses /mob/living/carbon/human/var/body_type to determine which gender sprite is drawn, so as to divorce it from the character's pronouns. It may seem unintuitive to use the mob's body_type instead of the bodypart's saved gender, but that's a conscious choice that shouldn't really matter much since it only affects the chest anyway, which can't be removed from the mob

Reply:
It affects some heads too, but I can easily replace limb_gender's assignment to use body type instead of gender.